### PR TITLE
ref 643132: server backup schedules

### PR DESCRIPTION
**Pull request by aeter**
Adds a `server` namespace and inside it, `stackit server backup-schedule` commands.

Testing Done:
* ran `make fmt`, `make lint`, `make test`
* tested the `stackit server backup-schedule` commands locally

### CODE REVIEW

File: internal/cmd/server/backup-schedule/delete/delete.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+			err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("delete Server Backup Schedule: %w", err)
+			}
+
+			operationState := "Deleted"
+			if model.Async {
+				operationState = "Triggered deletion of"
+			}
+			p.Info("%s server backup schedule %q\n", operationState, model.ScheduleId)
```
Comment (joaopalet) [inline internal/cmd/server/backup-schedule/delete/delete.go:74]: Is any of these operations async? Meaning that they return before finishing and then we need to check it's status through either a status field or until the get returns a 404?

Comment (joaopalet) [reply internal/cmd/server/backup-schedule/delete/delete.go:74]: If not then you do not need the `Triggered ...` part

Comment (joaopalet) [reply internal/cmd/server/backup-schedule/delete/delete.go:74]: This also applies for the `create` and `update` commands

File: internal/cmd/server/backup-schedule/update/update.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			currentBackupSchedule, err := apiClient.GetBackupScheduleExecute(ctx, model.ProjectId, model.ServerId, model.BackupScheduleId)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get current server backup schedule: %v", err)
+			}
```
Comment (joaopalet) [inline internal/cmd/server/backup-schedule/update/update.go:86]:
```suggestion
			if err != nil {
				p.Debug(print.ErrorLevel, "get current server backup schedule: %v", err)
                return err
			}
```

File: internal/cmd/server/backup-schedule/update/update.go [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+		Short: "Updates a Server Backup Schedule",
+		Long:  "Updates a Server Backup Schedule.",
+		Example: examples.Build(
+			examples.NewExample(
+				`Update the retention period of a server backup schedule for server "xxx" and schedule "zzz"`,
+				"$ stackit server backup-schedule update --server-id=xxx --backup-retention-period=20 zzz"),
+			examples.NewExample(
+				`Update the backup name of a server backup schedule for server "xxx" and schedule "zzz"`,
+				"$ stackit server backup-schedule update --server-id=xxx --backup-name=newname zzz"),
+		),
```
Comment (joaopalet) [inline internal/cmd/server/backup-schedule/update/update.go:67]:
```suggestion
		Example: examples.Build(
			examples.NewExample(
				`Update the retention period of the backup schedule "zzz" of server "xxx"`,
				"$ stackit server backup-schedule update zzz --server-id=xxx --backup-retention-period=20"),
			examples.NewExample(
				`Update the backup name of the backup schedule "zzz" of server "xxx"`,
				"$ stackit server backup-schedule update zzz --server-id=xxx --backup-name=newname"),
		),
```

File: internal/cmd/server/backup-schedule/enable/enable.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enables Server Backup Schedules for a project",
+		Long:  "Enables Server Backup Schedules for a project.",
+		Args:  args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Enable Server Backup Schedule functionality for your project`,
+				"$ stackit server backup-schedule enable"),
```
[the remark this answers is not in the document]
Comment (aeter) [reply internal/cmd/server/backup-schedule/enable/enable.go:38]: It enables them for a server. The API enables backups (and backup-schedules) for each server separately, with a `server_id` and `project_id` in the API url...

I'll add the `--server-id` to the example. Thank you!

**Pull request merged**